plugin/vcs/github/github.go

```diff
@@ -0,0 +1,286 @@
+package github
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+
+	"github.com/bytebase/bytebase/common"
+	"github.com/bytebase/bytebase/plugin/vcs"
+)
+
+func init() {
+	vcs.Register(vcs.GitHubCom, newProvider)
+}
+
+var _ vcs.Provider = (*Provider)(nil)
+
+// Provider is a GitHub.com VCS provider.
+type Provider struct {
+	l      *zap.Logger
+	client *http.Client
+}
+
+func newProvider(config vcs.ProviderConfig) vcs.Provider {
+	if config.Client == nil {
+		config.Client = &http.Client{}
+	}
+	return &Provider{
+		l:      config.Logger,
+		client: config.Client,
+	}
+}
+
+const apiURL = "https://api.github.com"
+
+func (p *Provider) APIURL(string) string {
+	return apiURL
+}
+
+// fetchUserInfo fetches user information from the given resourceURI, which
+// should be either "user" or "users/{username}".
+func (p *Provider) fetchUserInfo(ctx context.Context, oauthCtx common.OauthContext, resourceURI string) (*vcs.UserInfo, error) {
+	code, body, err := httpGet(
+		ctx,
+		p.client,
+		resourceURI,
+		&oauthCtx.AccessToken,
+		oauthContext{
+			ClientID:     oauthCtx.ClientID,
+			ClientSecret: oauthCtx.ClientSecret,
+			RefreshToken: oauthCtx.RefreshToken,
+		},
+		oauthCtx.Refresher,
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "GET")
+	}
+
+	if code == http.StatusNotFound {
+		errInfo := []string{"failed to fetch user info from GitHub.com"}
+		resourceURISplit := strings.Split(resourceURI, "/")
+		if len(resourceURI) > 1 {
+			errInfo = append(errInfo, fmt.Sprintf("UserID: %v", resourceURISplit[1]))
+		}
+		return nil, common.Errorf(common.NotFound, fmt.Errorf(strings.Join(errInfo, ", ")))
+	} else if code >= http.StatusMultipleChoices {
```

Comment:
nit: I personally prefer to use 300 here. It's a rare case that the magic number is more readable.